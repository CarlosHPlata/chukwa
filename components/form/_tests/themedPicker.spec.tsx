import { render, fireEvent } from "@testing-library/react-native";
import ThemedPicker from "../themedPicker";

describe("ThemedCheckbox", () => {
  it("should render the piicker correctly", () => {
    const values = [
      { value: "option1", label: "Option 1" },
      { value: "option2", label: "Option 2" },
    ];
    const renderTree = render(
      <ThemedPicker label="Select Option" values={values} />,
    );

    const picker = renderTree.getByTestId("themed-picker:Picker");
    expect(picker).toBeTruthy();
    expect(renderTree).toMatchSnapshot();
  });

  it("should handle value change events", () => {
    const values = [
      { value: "option1", label: "Option 1" },
      { value: "option2", label: "Option 2" },
    ];
    const mockOnValueChange = jest.fn();
    const renderTree = render(
      <ThemedPicker
        label="Select Option"
        values={values}
        onValueChange={mockOnValueChange}
      />,
    );

    const picker = renderTree.getByTestId("themed-picker:Picker");
    fireEvent(picker, "valueChange", "option2");

    expect(mockOnValueChange).toHaveBeenCalledWith("option2");
  });

  it("should display the correct label", () => {
    const values = [
      { value: "option1", label: "Option 1" },
      { value: "option2", label: "Option 2" },
    ];
    const renderTree = render(
      <ThemedPicker label="Select Option" values={values} />,
    );

    const label = renderTree.getByText("Select Option");
    expect(label).toBeTruthy();
  });
});
