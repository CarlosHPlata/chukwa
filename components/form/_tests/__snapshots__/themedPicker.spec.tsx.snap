// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedCheckbox should render the piicker correctly 1`] = `
[
  <Text>
    Select Option
  </Text>,
  <View
    style={
      {
        "marginBottom": 16,
      }
    }
  >
    <RNCPicker
      items={
        [
          {
            "label": "Option 1",
            "testID": undefined,
            "textColor": undefined,
            "value": "option1",
          },
          {
            "label": "Option 2",
            "testID": undefined,
            "textColor": undefined,
            "value": "option2",
          },
        ]
      }
      numberOfLines={1}
      onChange={[Function]}
      selectedIndex={0}
      style={
        [
          {
            "height": 216,
          },
          undefined,
        ]
      }
      testID="themed-picker:Picker"
    />
  </View>,
]
`;
